src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenDictionary.cs

```diff
@@ -41,7 +41,7 @@ public static class FrozenDictionary
         public static FrozenDictionary<TKey, TSource> ToFrozenDictionary<TSource, TKey>(
             this IEnumerable<TSource> source, Func<TSource, TKey> keySelector, IEqualityComparer<TKey>? comparer = null)
             where TKey : notnull =>
-            CreateFromDictionary(source.ToDictionary(keySelector, comparer));
+            source.ToDictionary(keySelector, comparer).ToFrozenDictionary(comparer);
```

Comment:
It seems `GetExistingFrozenOrNewDictionary` in line 31 is responsible for filtering empty inputs? Could you also add an assertion at the start of the `CreateFromDictionary` method?